src/bootstrap/doc.rs

```diff
@@ -463,7 +463,10 @@ impl Step for Std {
                 .arg("--resource-suffix")
                 .arg(&builder.version)
                 .arg("--index-page")
-                .arg(&builder.src.join("src/doc/index.md"));
+                .arg(&builder.src.join("src/doc/index.md"))
+                .arg("--extern-html-root-url")
+                .arg("std_detect=https://docs.rs/std_detect/0.1.5/")
```

Comment:
Version will eventually be changed, and this can lead to some hard to detect errors.

Reply:
This is exactly why I specify a version: otherwise older std docs version will always point to the latest `std_detect` version and that will very likely be completely wrong.

Reply:
Or we can check version of crate via metadata and fail if it not equal hardcoded one.

Reply:
We can extract the version from the libstd `Cargo.toml` file I guess...

Reply:
`cargo_metadata` crate should work, i guess.

Reply:
Why is std_detect not added to the list of crates we document?

Reply:
No idea. Should it be part of the crates documented?

Reply:
Maybe - it seems like a likely easy and correct way to expose it's documentation/source. There's a policy question of whether we *should* do that - it would probably be a reasonable stance that it's not part of the stable surface area so showing search results (for example) from it isn't appropriate. Perhaps rustdoc shouldn't render the src link in this case.

But I think this PR is almost certainly the wrong approach - if we want to not include it in search results but include links to it, there's probably a way to order the builds and later combine them or something like that to accomplish that.